packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -71,29 +73,43 @@ const emit = defineEmits<{
   validityChange: [state: ValidityState];
 }>();
 
-const isTouched = ref(false);
+const { t } = injectI18n();
 
-const inputElement = ref<HTMLInputElement | null>(null);
+const value = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
 
+const isTouched = ref(false);
+const inputElement = ref<HTMLInputElement | null>(null);
 const validityState = ref(inputElement.value?.validity);
 
 const errorMessage = computed(() => {
-  /* when the validity state is uninitialized or the form is valid, we don't show an error. */
   if (!validityState.value || validityState.value.valid) return "";
-  /* a custom error message always is considered first */
-  if (props.errorMessage) return props.errorMessage;
-
-  const element = inputElement.value;
-  /* when a language key is provided, we offer our own translations of the error messages 
-  to match the rest of the user's application */
-  if (props.lang && element) return useFormValidationMessage(element.validity, props);
-  /* we default to the browser's standard validation message that relies on the browser language */
-  return element?.validationMessage || "";
-});
 
-const value = computed({
-  get: () => props.modelValue,
-  set: (value) => emit("update:modelValue", value),
+  const errorType = getFirstInvalidType(validityState.value);
+  if (!errorType) return "";
+
+  // a custom error message always is considered first
+  if (props.errorMessage || errorType === "customError") return props.errorMessage;
+
+  // if the error is "typeMismatch", we will use an error message depending on the type property
+  if (errorType === "typeMismatch") {
+    const type = TRANSLATED_INPUT_TYPES.includes(props.type as TranslatedInputType)
+      ? (props.type as TranslatedInputType)
+      : "generic";
+    return t.value(`validations.typeMismatch.${type}`);
```

Comment:
![image](https://github.com/SchwarzIT/onyx/assets/151019977/be8bc715-b5af-4db4-a903-d02b177a62e8)

```suggestion
    return t.value(`validations.typeMismatch.${type}`, { value: value.value });
```